user_sync/connector/directory_ldap.py

```diff
@@ -269,35 +272,55 @@ def iter_users(self, users_filter):
                 if (last_attribute_name != None):
                     self.logger.warn('No email attribute: %s for dn: %s', last_attribute_name, dn)
                 continue
+
+            source_attributes = {}
             
             user = user_sync.connector.helper.create_blank_user()
             user['email'] = email
+            source_attributes['email'] = email
                 
             username, last_attribute_name = self.user_username_formatter.generate_value(record)
             if (username == None and last_attribute_name != None):
                 self.logger.info('No username attribute: %s for dn: %s', last_attribute_name, dn)    
             user['username'] = username if username != None else email
+            if (username != None):
+                source_attributes['username'] = username
                     
             domain, last_attribute_name = self.user_domain_formatter.generate_value(record)
             if (domain != None):
                 user['domain'] = domain
+                source_attributes['domain'] = domain
             elif (last_attribute_name != None):
                 self.logger.info('No domain attribute: %s for dn: %s', last_attribute_name, dn)    
                                                 
             given_name_value = LDAPValueFormatter.get_attribute_value(record, 'givenName')
             if (given_name_value != None):   
                 user['firstname'] = given_name_value
+                source_attributes['givenName'] = given_name_value
```

Comment:
The before state of target_attributes is exactly what you're asking for, isn't it?